src/ase/cw/model/OrderQueue.java

```diff
@@ -11,19 +11,44 @@
     private List<Order> loadedOrders;
     private OrderProducerListener listener;
     private int maxDelayTime = 1000;
+    private Thread orderQueueThread;
+    private boolean shouldPause = false;
+    private String name = "Order Queue";
+
 
     public OrderQueue(Vector<Order> loadedOrders, OrderProducerListener listener) {
         this.loadedOrders = loadedOrders;
         this.listener = listener;
+            if (orderQueueThread == null) {
+                orderQueueThread = new Thread(this);
+                orderQueueThread.setName(name);
+                orderQueueThread.start();
+            } else {
+                System.out.println(getName() + "already started");
+            }
+    }
+
+    public String getName() {
+        return name;
+
     }
 
     public void setMaxDelayTime(int delay) {
         this.maxDelayTime = delay;
     }
 
+
     @Override
     public void run() {
         for (Order order : this.loadedOrders) {
+            synchronized (orderQueueThread) {
```

Comment:
you should only synchronize the first while loop, not the whole loop up to Thread.sleep